### PR TITLE
#162949214 Add social login

### PR DESCRIPTION
**Pull request by kwanj-k**
#### What does this PR do?

Add social authentication to the API

#### Description of Task to be completed?

* Enable Google, Facebook and Twitter login
* Add tests

#### How should this be manually tested?

1. Clone the repo: `git clone https://github.com/andela/ah-the-unsullied.git`
2. Cd into the cloned repo 
3. Git pull branch ft-social-auth-162949214
4. Set up a virtual environment and install requirements:
      ```
            python3 -m venv venv
            source venv/bin/activate
            pip3 install -r requirements.txt

     ```
5. Set up a .env file in the same level as the venv file and add the following:
      ```
           source venv/bin/activate
           export SECRET_KEY=""
           export DATABASE_URL=""
           export DJANGO_SETTINGS_MODULE='authors.settings'
           export CLOUDINARY_NAME=''
           export CLOUDINARY_KEY=''
           export CLOUDINARY_SECRET=''
           export HOST_PASSWORD=""
           export HOST_USER=""
           export SMTP_HOST=''
           export EMAIL_PORT=''
           export FACEBOOK_KEY='
           export FACEBOOK_SECRET=''
           export TWITTER_KEY=''
           export TWITTER_SECRET_KEY=''
           export GOOGLE_KEY=''
           export GOOGLE_SECRET=''
      ```
6. Then run: 
     ``` 
            source .env 
            ./manage.py migrate
            ./manage.py runserver
      ```
7. To test locally, you need an access token and add it to the postman body:
      ```
            {
               "provider":"",
               "access_token":""
                }

      ```
#### Any background context you want to provide?
You need the above keys from the admin.

#### What are the relevant pivotal tracker stories?
[162949206](#162949206)
#### Screenshots (if appropriate)
None
#### Questions:

### CODE REVIEW

**Review by kelvinwm — DISMISSED** [review]

**Review by MbugwaSami — DISMISSED** [review]
looks good to me

**Review by nesh-dev — DISMISSED** [review]

File: authors/tests/test_social_auth.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+class TestSocialAuth(TestBase):
+    """This class has tests social auth."""
+
+    def test_invalid_token(self):
+        data = self.invalid_token
+        url = self.social_auth_url
+        response = self.client.post(url, data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_inalid_provider(self):
```
Comment (Allan690) [inline authors/tests/test_social_auth.py]: Kindly resolve this typo

File: authors/tests/test_social_auth.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import json
+import os
+from rest_framework.views import status
+from django.urls import reverse
```
Comment (Allan690) [inline authors/tests/test_social_auth.py]: You have an unused import here

File: authors/tests/test_social_auth.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import json
```
Comment (Allan690) [inline authors/tests/test_social_auth.py]: Another unused import

File: authors/tests/test_register.py [hunk 4]
```diff
@@ -106,3 +106,20 @@ def test_user_update(self):
             format='json'
             )
         self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_get_current_user(self):
```
Comment (Allan690) [inline authors/tests/test_register.py:110]: Kindly add a docstring to this test

**Review by Allan690 — DISMISSED** [review]

**Review by Allan690 — DISMISSED** [review]

**Review by Allan690 — APPROVED** [review]

**Review by nesh-dev — APPROVED** [review]

**Review by indungu — APPROVED** [review]

**Pull request merged**